runtime/memory.c

```diff
@@ -419,7 +419,7 @@ CAMLexport void caml_set_fields (value obj, value v)
 }
 
 Caml_inline value alloc_shr(mlsize_t wosize, tag_t tag, reserved_t reserved,
-                            int noexc)
+                            int noexc, int longlived)
```

Comment:
This change is not necessary for this PR, and I don't think that's the API we want in the long term, so it would be better to remove it.

Reply:
My intuition is that we want an unmarshalling entry-point where the user can that explicit say whether the data will be short-lived and long-lived. (Independently of maybe having another, more global API to indicate ramp-up modes.) I think that allow thing will require a flag for `alloc_shr`. But I'm happy to remove the flag for now to keep the first iteration simple.